packages/apollo-server-core/src/runHttpQuery.ts

```diff
@@ -351,21 +370,67 @@ export async function runHttpQuery(
         }),
       };
     }
-  }) as Array<Promise<ExecutionResult>>;
+  }) as Array<Promise<ExecutionResult & { extensions?: Record<string, any> }>>;
 
   const responses = await Promise.all(requests);
 
+  const responseInit: ApolloServerHttpResponse = {
+    headers: {
+      'Content-Type': 'application/json',
+    },
+  };
+
+  //enabling cacheControl means that the user would like the cache-control
+  //extensions we are running the proxy, so we should not strip out the cache
+  //control extension.
+  if (!optionsObject.cacheControl) {
+    responseInit.headers = {
+      ...responseInit.headers,
+      ...calculateCacheControlHeaders(responses),
```

Comment:
Shouldn't we do this even when you explicitly ask for cacheControl? Perhaps there is a second option to cacheControl, eg cacheControl.setHTTPHeaders, which default to true.